### ISSUE TITLE
Error response from creating a new channel could be more helpful

### ISSUE THREAD

**Issue by candlerb** [zatosource/zato#429]
(Possibly related to #418)

I tried to create a SOAP channel like this:

* `Connections > Channels > SOAP`
* Create a new SOAP channel
* Name: tt-dev.api1
* Active: [X]
* URL path: /tt-dev/api1
* Method: POST
* Data format: XML
* SOAP action: [leave blank]
* SOAP version: 1.1
* Service: srv-nd1651.receive-message
* Security definition: No security definition

This resulted in the following unhelpful error within the browser (exception all one one line, reformatted here for clarity)

~~~
Could not create the object, e:[Traceback (most recent call last):
  File "/opt/zato/2.0.3/zato-web-admin/src/zato/admin/web/views/http_soap.py", line 189, in create
    response = req.zato.client.invoke('zato.http-soap.create', _get_edit_create_message(req.POST))
  File "/opt/zato/2.0.3/zato-web-admin/src/zato/admin/middleware.py", line 70, in invoke
    raise Exception(response.inner.text)
Exception: {"zato_env": {"details": "Traceback (most recent call last):\n
File \"\/opt\/zato\/2.0.3\/zato-server\/src\/zato\/server\/connection\/http_soap\/channel.py\", line 198, in dispatch\n
    payload, worker_store, self.simple_io_config, post_data)\n
  File \"\/opt\/zato\/2.0.3\/zato-server\/src\/zato\/server\/connection\/http_soap\/channel.py\", line 351, in handle\n
    params_priority=channel_item.params_pri)\n
  File \"\/opt\/zato\/2.0.3\/zato-server\/src\/zato\/server\/service\/__init__.py\", line 423, in update_handle\n
    raise e\nZatoException: Expected elem:[id] not found in item:[{u'id': u'', u'name': u''}]\n",
"result": "ZATO_ERROR", "cid": "K075QVVFDNT2B5KS01CMN2V3BMWG"}}
]
~~~

It took grubbing about in logs to find out what was going on. There is actually a large amount of log spew created for this event:

~~~
2015-04-21 17:14:50,896 - ESC[1;31mERRORESC[0m - 6369:Dummy-201 - zato.http-soap.create:22 - Service
 [srv-nd1651.receive-message] does not exist on this cluster
2015-04-21 17:14:50,899 - ESC[1;33mWARNINGESC[0m - 6369:Dummy-201 - zato.server.service:22 - Traceba
ck (most recent call last):
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 402, in update_handle
    self._invoke(service, channel)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 345, in _invoke
    service.handle()
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/internal/http_soap.py", line 150, in han
dle
    raise Exception(msg)
Exception: Service [srv-nd1651.receive-message] does not exist on this cluster

2015-04-21 17:14:50,902 - ESC[1;33mWARNINGESC[0m - 6369:Dummy-201 - zato.server.service:22 - Could n
ot invoke `zato.http-soap.create`, e:`Traceback (most recent call last):
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 467, in invoke_by_imp
l_name
    return self.update_handle(*invoke_args, **kwargs)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 414, in update_handle
    response = set_response_func(service, data_format=data_format, transport=transport, **kwargs)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 309, in set_response_
data
    response = response.getvalue(serialize=kwargs['serialize'])
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/reqresp/__init__.py", line 344, in getvalue
    elem_value = self._getvalue(name, item, is_sa_namedtuple, is_required, leave_as_is)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/reqresp/__init__.py", line 290, in _getvalue
    raise ZatoException(self.zato_cid, msg)
ZatoException: Expected elem:[id] not found in item:[{u'id': u'', u'name': u''}]
`
2015-04-21 17:14:50,904 - ESC[1;33mWARNINGESC[0m - 6369:Dummy-201 - zato.server.service:22 - Traceback (most recent call last):
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 402, in update_handle
    self._invoke(service, channel)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 345, in _invoke
    service.handle()
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/internal/service.py", line 271, in handle
    response = func(id_, payload, channel, data_format, transport, serialize=True)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 483, in invoke
    return self.invoke_by_impl_name(self.server.service_store.name_to_impl_name[name], *args, **kwargs)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 467, in invoke_by_impl_name
    return self.update_handle(*invoke_args, **kwargs)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 414, in update_handle
    response = set_response_func(service, data_format=data_format, transport=transport, **kwargs)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 309, in set_response_data
    response = response.getvalue(serialize=kwargs['serialize'])
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/reqresp/__init__.py", line 344, in getvalue
    elem_value = self._getvalue(name, item, is_sa_namedtuple, is_required, leave_as_is)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/reqresp/__init__.py", line 290, in _getvalue
    raise ZatoException(self.zato_cid, msg)
ZatoException: Expected elem:[id] not found in item:[{u'id': u'', u'name': u''}]
2015-04-21 17:14:50,905 - ESC[1;31mERRORESC[0m - 6369:Dummy-201 - zato.server.connection.http_soap.channel:22 - Caught an exception, cid:[K07CK2HRSPP7EPGTRXVMYWV3CT6X], status_code:[500], _format_exc:[Traceback (most recent call last):
  File "/opt/zato/2.0.3/zato-server/src/zato/server/connection/http_soap/channel.py", line 198, in dispatch
    payload, worker_store, self.simple_io_config, post_data)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/connection/http_soap/channel.py", line 351, in handle
    params_priority=channel_item.params_pri)
  File "/opt/zato/2.0.3/zato-server/src/zato/server/service/__init__.py", line 423, in update_handle
    raise e
ZatoException: Expected elem:[id] not found in item:[{u'id': u'', u'name': u''}]
]
~~~

Working backwards from the end of the log, I found the actual error which is simply `Service
 [srv-nd1651.receive-message] does not exist on this cluster` - i.e. I had given the channel the wrong service name, which is a basic mistake.

It seems that the true exception, although logged, does not propagate back to the web GUI user.

**Comment by dsuch** [comment]
Thanks @candlerb - I tagged it with 2.0.4 so it will be seen to soon.

In 2.1, and already in master since #384, you will be given a select list to pick a service from, cutting down on the number of such exceptions.

**Comment by dsuch** [comment]
Done in aee7bc9d2dd6af95714fcdebca462b64ee7c22fa. Backported to support/2.0. Will be released in 2.0.4.
